zetaclient/bitcoin/bitcoin_client.go

```diff
@@ -1331,23 +1352,29 @@ func (ob *BTCChainClient) checkTSSVout(params *types.OutboundTxParams, vouts []b
 	nonce := params.OutboundTxTssNonce
 	tssAddress := ob.Tss.BTCAddress()
 	for _, vout := range vouts {
-		recvAddress, amount, err := DecodeP2WPKHVout(vout, ob.chain)
+		// decode receiver and amount from vout
+		receiverExpected := tssAddress
+		if vout.N == 1 {
```

Comment:
Maybe a switch for Vout.N might clean this up a bit. Then you can have a default for indices greater than 3

Reply:
Good idea. There will be no default as I have the length check of `if !(len(vouts) == 2 || len(vouts) == 3)` at the beginning.

Reply:
https://github.com/zeta-chain/node/pull/1942/commits/f02cc5253a5fbc6f0402c41ad0905837de3eb2c9